### PR TITLE
Feature/app 3230

### PR DESCRIPTION
**Pull request by Werder630**

### CODE REVIEW

File: src/main/java/com/yandex/money/api/model/Operation.java [hunk 1]
```diff
@@ -162,6 +163,16 @@
...
 
     /**
+     * Id of categories
+     */
+    public final List<Integer> categories;
+
+    /**
+     * Type of showcase
+     */
+    public final String format;
```
Comment (yasevich) [inline src/main/java/com/yandex/money/api/model/Operation.java]: The type must be `ShowcaseReference.Format`.

File: src/main/java/com/yandex/money/api/model/Operation.java [hunk 2]
```diff
@@ -298,6 +314,8 @@ public int hashCode() {
         result = 31 * result + (favorite != null ? favorite.hashCode() : 0);
         result = 31 * result + type.hashCode();
         result = 31 * result + (digitalGoods != null ? digitalGoods.hashCode() : 0);
+        result = 31 * result + (categories != null ? categories.hashCode() : 0);
```
Comment (yasevich) [inline src/main/java/com/yandex/money/api/model/Operation.java]: Field `categories` can never be `null`.

File: src/main/java/com/yandex/money/api/model/Operation.java [hunk 3]
```diff
@@ -574,6 +594,16 @@ public Builder setDigitalGoods(DigitalGoods digitalGoods) {
             return this;
         }
 
+        public Builder setCategories(List<Integer> categories) {
+            this.categories = checkNotNull(categories, "categories");
```
Comment (yasevich) [inline src/main/java/com/yandex/money/api/model/Operation.java]: All model consistency checks must be done in a constructor.

File: src/main/java/com/yandex/money/api/model/Operation.java (hunk 1 above)
Comment (Werder630) [inline src/main/java/com/yandex/money/api/model/Operation.java]: Done

File: src/main/java/com/yandex/money/api/model/Operation.java (hunk 2 above)
Comment (Werder630) [inline src/main/java/com/yandex/money/api/model/Operation.java]: Done

File: src/main/java/com/yandex/money/api/model/Operation.java (hunk 3 above)
Comment (Werder630) [inline src/main/java/com/yandex/money/api/model/Operation.java]: Done

**Pull request closed without merging**